README.md

```diff
@@ -1 +1,1146 @@
+           @kubernetes:Service                      - complete 1/1
+           @kubernetes:Ingress                      - complete 1/1
+           @kubernetes:Deployment                   - complete 1/1
+           @kubernetes:Docker                       - complete 3/3 
+
+           Run following command to deploy kubernetes artifacts: 
+           kubectl apply -f ./target/message_transformation/kubernetes/
+```
+
+- You can verify that the docker image that we specified in `` @kubernetes:Deployment `` is created, by using `` docker images ``. 
+- Also the Kubernetes artifacts related our service, will be generated in `` ./target/message_transformation/kubernetes``. 
+- Now you can create the Kubernetes deployment using:
+
+```
+   $kubectl apply -f ./target/message_transformation/kubernetes 
+
+```
+
+- You can verify Kubernetes deployment, service and ingress are running properly, by using following Kubernetes commands. 
+
+```
+   $kubectl get service
+   $kubectl get deploy
+   $kubectl get pods
+   $kubectl get ingress
+```
+
+- If everything is successfully deployed, you can invoke the service either via Node port or ingress. 
+
+Node Port:
+ 
+```
+   curl -v http://localhost:9090/contentfilter -d '{"id" : 105, "name" : "ballerinauser", "city" : "Colombo 03", "gender" : "male"}' -H "Content-Type:application/json" -X POST 
+```
+
+Ingress:
+
+Add `/etc/hosts` entry to match hostname. 
+
+``` 
+   127.0.0.1 ballerina.guides.io
+```
+
+Access the service 
+
+``` 
+    curl -v http://localhost:9090/contentfilter -d '{"id" : 105, "name" : "ballerinauser", "city" : "Colombo 03", "gender" : "male"}' -H "Content-Type:application/json" -X POST
+```
+## Observability 
+Ballerina is by default observable. Meaning you can easily observe your services, resources, etc. Refer to [how-to-observe-ballerina-code](https://ballerina.io/learn/how-to-observe-ballerina-code/) for more information.
+However, observability is disabled by default via configuration. Observability can be enabled by adding following configurations to `ballerina.conf` file and starting the ballerina service using it.
+
+```ballerina
+[b7a.observability]
+
+[b7a.observability.metrics]
+# Flag to enable Metrics
+enabled=true
+
+[b7a.observability.tracing]
+# Flag to enable Tracing
+enabled=true
+```
+> **NOTE**: The above configuration is the minimum configuration needed to enable tracing and metrics. With these configurations, default values are loaded as the other configuration parameters of metrics and tracing.
+
+### Tracing
+You can monitor Ballerina services using inbuilt tracing capabilities of Ballerina. Let's use [Jaeger](https://github.com/jaegertracing/jaeger) as the distributed tracing system.
+
+Follow the steps below to use tracing with Ballerina.
+
+You can add the following configurations for tracing. Note that these configurations are optional if you already have the basic configuration in `ballerina.conf` as described above.
```

Comment:
Shall we explain exactly what these configurations do?

Reply:
Done